src/odc/nc2odb.py

```diff
@@ -0,0 +1,245 @@
+
+        Dates.append(Date)
+        Times.append(Time)
+
+    return (Dates, Times)
+
+
+def ConvertDtRefOffset(RefDtInt, OffsetTime):
+    # convert the reference date time to a datetime object
+    TempRef = RefDtInt
+    Year = TempRef // 1000000
+    TempRef = TempRef % 1000000
+    Month = TempRef // 10000
+    TempRef = TempRef % 10000
+    Day = TempRef // 100
+    Hour = TempRef % 100
+    RefDt = dt.datetime(Year, Month, Day, Hour)
+
+    Dates = []
+    Times = []
+    for Otime in OffsetTime:
+        OffsetSeconds = round(3600 * Otime)
+        OffsetDt = dt.timedelta(seconds=OffsetSeconds)
+        Dt = RefDt + OffsetDt
+        (Date, Time) = DtToDateTime(Dt)
+
+        Dates.append(Date)
+        Times.append(Time)
+
+    return (Dates, Times)
+
+
+def DtToDateTime(Dt):
+    Date = Dt.year * 10000
+    Date += Dt.month * 100
+    Date += Dt.day
+
+    Time = Dt.hour * 10000
+    Time += Dt.minute * 100
+    Time += Dt.second
+
+    return (Date, Time)
+
+
+###################################################################################
+# MAIN
+###################################################################################
+ScriptName = os.path.basename(sys.argv[0])
+
+# Parse command line
+ap = argparse.ArgumentParser()
+ap.add_argument("input_netcdf", help="path to input netCDF4 file")
+ap.add_argument("output_odb", help="path to output ODB file")
+ap.add_argument("-c", "--clobber", action="store_true",
+                help="allow overwrite of output ODB file")
+ap.add_argument("-k", "--keep_csv", action="store_true",
+                help="do not delete the temporary CSV file")
+
+MyArgs = ap.parse_args()
+
+NetcdfFname = MyArgs.input_netcdf
+OdbFname = MyArgs.output_odb
+ClobberOfile = MyArgs.clobber
+KeepCsv = MyArgs.keep_csv
+
+# Check files
+BadArgs = False
+if (not os.path.isfile(NetcdfFname)):
+    print("ERROR: {0:s}: Specified input netCDF4 file does not exist: {1:s}".format(
+        ScriptName, NetcdfFname))
+    print("")
+    BadArgs = True
+
+if (os.path.isfile(OdbFname)):
+    if (ClobberOfile):
+        print("WARNING: {0:s}: Overwriting ODB file: {1:s}".format(
+            ScriptName, OdbFname))
+        print("")
+    else:
+        print("ERROR: {0:s}: Specified ODB file already exists: {1:s}".format(
+            ScriptName, OdbFname))
+        print("ERROR: {0:s}:   Use -c option to overwrite.".format(ScriptName))
+        print("")
+        BadArgs = True
+
+if (BadArgs):
+    sys.exit(2)
+
+# Arguments are okay. Read in the odb file using the xarray interface. Then covert to
+# a pandas dataframe, write out the pandas into the odb file.
+
+# Select the variables that are to be transferred to the output which are all the ones
+# with nlocs as the first dimension. Also change datetime@MetatData to two integer
+# variables: date@MetaData and time@MetaData (with formats YYYYMMDD, hhmmss respectively).
+NcDs = nc.Dataset(NetcdfFname)
+IntMissingValue = 2147483647
+DoubleMissingValue = -2147483647
```

Comment:
shouldn't this be -2147483647. 0 or float(IntMissingValue) ?

Reply:
Currently, the script is using the `odc import` command to build the output ODB file. The import command wants a CSV file for input and the integer style missing value for a variable of type double is working. Plus, using an integer value makes it simpler to get all the digits to come out correctly compared to using a float value.

It would probably be good to eventually replace the `odc import` method with methods in odyssey (odc python api) so we can avoid creating giant CSV files. However, I would like to defer this to a subsequent PR.

Reply:
Ok. I defer to the inner workings of `odc import`.